x-pack/plugin/rank-vectors/src/test/java/org/elasticsearch/xpack/rank/vectors/mapper/RankVectorsFieldTypeTests.java

```diff
@@ -86,9 +88,12 @@ public void testDocValueFormat() {
 
     public void testFetchSourceValue() throws IOException {
         RankVectorsFieldType fft = createFloatFieldType();
-        List<List<Double>> vector = List.of(List.of(0.0, 1.0, 2.0, 3.0, 4.0, 6.0));
-        assertEquals(vector, fetchSourceValue(fft, vector));
+        List<List<Double>> vectorFromXContent = List.of(List.of(0.0, 1.0, 2.0, 3.0, 4.0, 6.0));
+        List<float[]> vector = List.of(new float[] { 0.0f, 1.0f, 2.0f, 3.0f, 4.0f, 6.0f });
+        assertThat(fetchSourceValue(fft, vectorFromXContent), iterableWithSize(1));
+        assertThat(fetchSourceValue(fft, vectorFromXContent).get(0), equalTo(vector.get(0)));
         RankVectorsFieldType bft = createByteFieldType();
-        assertEquals(vector, fetchSourceValue(bft, vector));
+        assertThat(fetchSourceValue(bft, vectorFromXContent), iterableWithSize(1));
+        assertThat(fetchSourceValue(bft, vectorFromXContent).get(0), equalTo(vector.get(0)));
```

Comment:
```suggestion
        assertThat(fetchSourceValue(bft, vectorFromXContent).get(0), equalTo(vector.get(0)));
                String hexStr = HexFormat.of().formatHex(new byte[] { 0, 1, 2, 3, 4, 6 });
        List<String> hexVecs = List.of(hexStr);
        assertThat(fetchSourceValue(bft, hexVecs), iterableWithSize(1));
        assertThat(fetchSourceValue(bft, hexVecs).get(0), equalTo(hexStr));
```
Something like this to test the hexstring.